backend/src/main/java/de/neuefische/backend/service/UserService.java

```diff
@@ -37,4 +38,18 @@ public User findUserById(String id){
                 .orElseThrow(() -> new NoSuchElementException("User with id " + id + "does not exsist"));
     }
 
+    public User updateUser(String id, User editedUser){
+        checkIfExsists(id);
+        return userRepo.save(editedUser);
+    }
+
+    private void checkIfExsists(String id) throws NoSuchUserException {
+        for (User user :
+                listUsers()) {
+            if (user.id().equals(id)) {
+                return;
+            }
+        }
```

Comment:
Load all user just to check if one exists sound pretty expensive...
I would use the appropriate mongodb method.